### PR TITLE
Update build metadata on deletion, fix misc bugs

### LINKED ISSUE #560: [BUG] - Delete environment does not change storage size
### Describe the bug

When we go through the delete env process, we see the process "succeed", but the total storage size does not change. 

```
[2023-09-05 21:11:10,476: INFO/MainProcess] Task task_delete_build[d12572cf-694d-4341-86ab-34a21dac779f] received
[2023-09-05 21:11:10,482: WARNING/ForkPoolWorker-2] [CondaStoreWorker] deleting artifacts for build=89
[2023-09-05 21:11:13,712: INFO/ForkPoolWorker-2] Task task_delete_build[d12572cf-694d-4341-86ab-34a21dac779f] succeeded in 3.235292388126254s: None
```

We're seeing this on JATIC and @sblair-metrostar has seen this on Onyx. 

### Expected behavior

I expect when I go through the deletion process on the UI or otherwise, for the process to remove the environment from disk completely.

### How to Reproduce the problem?

* Create a big environment
* note the total storage size (via the admin interface)
* Delete the environment
* Wait a big for the deletion process to complete
* note the total storage size hasn't changed. 

I'm sure there is a better method for those with a local dev deployment of conda-store. This is just from the UI. 

### Output

_No response_

### Versions and dependencies used.

_No response_

### Anything else?

Related: https://github.com/conda-incubator/conda-store/issues/420

### PR DESCRIPTION
**Pull request by nkaretnikov**
Fixes #560.

### CODE REVIEW

File: conda-store-server/conda_store_server/worker/tasks.py [hunk 1]
```diff
@@ -231,7 +232,7 @@ def delete_build_artifact(db: Session, conda_store, build_artifact):
         # ignore key
         conda_prefix = build_artifact.build.build_path(conda_store)
         # be REALLY sure this is a directory within store directory
-        if conda_prefix.startswith(conda_store.store_directory) and os.path.isdir(
+        if str(conda_prefix).startswith(conda_store.store_directory) and os.path.isdir(
```
Comment (nkaretnikov) [inline conda-store-server/conda_store_server/worker/tasks.py:235]: Bug fix: `PosixPath` object (`conda_prefix`) has no attribute `startswith`, so need to convert to `str` first

File: conda-store-server/conda_store_server/worker/tasks.py [hunk 2]
```diff
@@ -254,14 +255,20 @@ def task_delete_build(self, build_id):
...
             excluded_artifact_types=settings.build_artifacts_kept_on_deletion,
         ).all():
             delete_build_artifact(db, conda_store, build_artifact)
-        conda_store.db.commit()
+
+        # Updates build size and marks build as deleted
+        build.deleted_on = datetime.datetime.utcnow()
+        build.size = 0
+
+        db.commit()
```
Comment (nkaretnikov) [inline conda-store-server/conda_store_server/worker/tasks.py:271]: Bug fix: `CondaStore` object has no attribute `db`, need to just use `db` for this

File: conda-store-server/conda_store_server/worker/tasks.py [hunk 3]
```diff
@@ -254,14 +255,20 @@ def task_delete_build(self, build_id):
...
             db,
             build_id=build_id,
             excluded_artifact_types=settings.build_artifacts_kept_on_deletion,
         ).all():
             delete_build_artifact(db, conda_store, build_artifact)
-        conda_store.db.commit()
+
+        # Updates build size and marks build as deleted
+        build.deleted_on = datetime.datetime.utcnow()
+        build.size = 0
```
Comment (nkaretnikov) [inline conda-store-server/conda_store_server/worker/tasks.py:269]: Metadata fix: size value of the current build is shown in the web UI as `size`. Need to update that when deleting artifacts here.

How it works in more detail:

```html
{% for environment in environments %}
<div class="card my-2">
    <div class="card-body">
        <h5 class="card-title">
            <a href="{{ url_for('ui_get_environment', namespace=environment.namespace.name, environment_name=environment.name) }}">{{ environment.namespace.name }} / <span class="badge badge-info">{{ environment.name }}</span></a>
            <span class="badge badge-light">{{ (environment.current_build.size or 0) | filesizeformat(true) }}</span>
        </h5>
```

=>

```html
 document.querySelectorAll('button[data-action="delete"]').forEach(item => {
     item.addEventListener('click', (event) => deleteBuild(event));
 })
```

=>

```html
 async function deleteBuild(event) {
...
     let url = `{{ url_for('api_list_builds') }}${ buildId }/`;
     let response = await fetch(url, { method: 'DELETE' });
...
```

=>

```py
@router_api.delete(
    "/build/{build_id}/",
    response_model=schema.APIAckResponse,
)
async def api_delete_build(
...
        conda_store.delete_build(db, build_id)
```

=>

```py
     def delete_build(self, db: Session, build_id: int):
...
        tasks.task_delete_build.si(build.id).apply_async()
```

=>

```py
task_delete_build (this function)
```

**Comment by nkaretnikov** [comment]
To test that this works:

- Remove all DB state files from before if you have any
- Log in
- Create this env via `http://localhost:5000/admin/create/`:
```
channels:
- conda-forge
dependencies:
- rust>=1.71.1
- binutils-meta>=1.0.4
- gcc>=12.2.0
- llvm-meta>=13.0.1
- cudatoolkit-dev>=11.4.0
description: slow env
name: slow
prefix: null
variables: null
```
- Go to `http://localhost:5000/admin/environment/default/slow/` and wait for the build to complete (takes around 5 mins)
- Once complete, see if the size on the same page has updated. For me, it shows `default/slow 7.3 GiB`
- Delete Build 1 (this build) by clicking on the Trash Bin icon on the same page
- Web UI should immediately update to `default/slow 0 Bytes`
- Click on the Rebuild icon on the same page to build this env again
- Wait 5 mins or so for Build 2 to complete
- Once done, the Web UI should show `default/slow 7.3 GiB` again.

**Comment by pierrotsmnrd** [comment]
Tested locally, and code reviewed : It works, with [PR 574](https://github.com/conda-incubator/conda-store/pull/574) being merged. So merge 574 before merging this one.

**Review by pierrotsmnrd — APPROVED** [review]

**Pull request merged**